generators/openshift/files.js

```diff
@@ -10,9 +10,11 @@ function writeFiles() {
             for (let i = 0; i < this.appConfigs.length; i++) {
                 const appName = this.appConfigs[i].baseName.toLowerCase();
                 this.app = this.appConfigs[i];
-                this.template('_deployment.yml', `${this.directoryPath}/ocp/${appName}/${appName}-deployment.yml`);
+                this.template('_deployment.yml', `${this.directoryPath}/ocp/${appName}/${appName}-deployment.yml`); //microsservice
+                this.template('_deployment.yml', `src/main/docker/openshift/${appName}-deployment-template.yml`); //monolithic
+                this.template('gradle/_openshift.gradle', `gradle/openshift.gradle`);
 
-                if (this.app.prodDatabaseType !== 'no') {
+                if (this.app.prodDatabaseType !== 'no' && this.app.prodDatabaseType !== 'mssql') {
```

Comment:
This comes from main generator. MSSQL is already there as one of the prod db types. All you have to do is to add a file under /templates/db/_mssql.yml. These changes are not really required. Just follow how it is done for other DB types like MySQL, Postgres (these options come from the main generator)